external/llvm-project/mlir/lib/Conversion/TosaToLinalg/TosaToLinalgNamed.cpp

```diff
@@ -199,6 +200,22 @@ class ConvConverter : public OpConversionPattern<TosaConvOp> {
     DenseI64ArrayAttr strideTosaAttr = op.getStrideAttr();
     DenseI64ArrayAttr dilationTosaAttr = op.getDilationAttr();
     bool isQuantized = op.getQuantizationInfo().has_value();
+    int64_t group = 1;
+
+    if (auto convop = dyn_cast<tosa::Conv2DOp>(&op)) {
+      if (convop->getGroup().has_value())
+        group = convop->getGroup().value();
+    }
+
+    if (group > 1 && isConv2DOp &&
+        !std::is_same<LinalgConvOp, linalg::Conv2DNhwgcGfhwcOp>::value)
+      return rewriter.notifyMatchFailure(
+          op, "tosa.conv ops should map to grouped convolution ops");
```

Comment:
I find this error message to be a bit cryptic.
Can we say something like "grouped tosa.conv ops should map to linalg::<specific> op" 

Please see my other comment above. I think we can do a transpose and not introduce a new linalg op.
Or at least I'd like to understand why such an approach wont work for our requirements.

Reply:
I don't want this PR to minimally fit our requirements but to be something acceptable to MLIR as a whole.

Reply:
if we are eagerly upstreaming these, then it is fine.

 > but to be something acceptable to MLIR as a whole.

Yea, my comments were based off that this is just going to be rocMLIR thing downstream.

Its a shame that upstream cant fold linalg transposes when they are creating linalg.generic ops where I/O affine maps could nicely handle the folding. So we have to keep adding all permutations of layouts as linalg named ops.